Cargo.toml

```diff
@@ -7,6 +7,7 @@ homepage = "https://crates.io"
 repository = "https://github.com/rust-lang/cargo"
 documentation = "https://docs.rs/cargo"
 readme = "README.md"
+rust-version = "1.58"
```

Comment:
Unless we automate the verification of our `rust-version`, I'm not a fan of setting it as its likely to become inaccurate, like it is in this PR at the time of my writing.

iirc it will automatically be set in the published crate to the version used when publishing (which will be valid if the user doesn't pass `--no-verify`).

Reply:
If it’s supposed to be set at the moment, it isn’t, as I stepped on a giant rake just today.

Reply:
And that’s the thing, I created this PR to get feedback, because I can’t possibly go through everything and pinpoint the exact version. It failed with 1.57.0 because of a specific change, so that’s what I put.

Reply:
Automating should be trivial, get the version, use rustup to install the appropriate toolchain and build. But, again, you’re the reviewer and you should know if that’s feasible.